path-manager/app/services/PathStore.scala

```diff
@@ -96,7 +96,72 @@ object PathStore extends Logging {
     }
   }
 
-  def updateCanonical(newPath: String, id: Long) = {
+  def registerAlias(proposedAliasPathRecord: PathRecord) = {
+
+    val id = proposedAliasPathRecord.identifier
+
+    logger.debug(s"Registering path [${proposedAliasPathRecord.path }] for [$id]")
+
+    if (!PathValidator.isValid(proposedAliasPathRecord.path)) {
+      Left(s"invalid path [${proposedAliasPathRecord.path}]")
+    } else {
+      val existingPath = Option(Dynamo.pathsTable.getItem("path", proposedAliasPathRecord.path)).map(PathRecord(_))
+
+      existingPath match {
+        case Some(pr) if (pr.identifier != id) => {
+          logger.warn(s"Failed to register existing path [${proposedAliasPathRecord.path}], already claimed by id [${pr.identifier}], submitting id [$id]")
+          Left("path already in use")
+        }
+        case _ => {
+          putPathItemAndAwaitIndexUpdate(proposedAliasPathRecord)
+
+          logger.debug(s"Registered new alias path [${proposedAliasPathRecord.path}] for id [$id] successfully")
+          Right(List(proposedAliasPathRecord).groupBy(_.`type`))
+        }
+      }
+    }
+  }
+
+  def updateCanonicalWithAlias(newPath: String, id: Long): Either[String, Map[String, List[PathRecord]]] = {
+    //This takes the canonical path and makes it an alias. And then adds a new canonical path for newPath.
+    
+    logger.debug(s"Updating canonical path [$newPath}] for [$id] and creating alias to old path.")
+
+    if (!PathValidator.isValid(newPath)) {
+      Left(s"invalid path [$newPath]")
```

Comment:
[minor] since we have an if/else block (and this is not the guard pattern of early returning) inverting the if/else and removing the `!` negation would make this easier to follow

Reply:
alternatively we could add a new `isInvalid` method to `PathValidator` (a one liner that does the negation there)